jetbrains-core/tst/software/aws/toolkits/jetbrains/services/ecs/exec/RunCommandDialogTest.kt

```diff
@@ -51,25 +87,92 @@ class RunCommandDialogTest {
+            val execCommandParameters = RunCommandDialog(projectRule.project, container, connectionSettings).constructExecCommandParameters(command)
             assertThat(execCommandParameters).isEqualTo(verifyCommand)
         }
     }
 
     @Test
-    fun `Run Parameters of Execute Command are set correctly`() {
+    fun `Credentials are attached as environment variables when running AWS CLI`() {
         resourceCache.addEntry(
             projectRule.project, EcsResources.listTasks(clusterArn, serviceArn),
             CompletableFuture.completedFuture(taskList)
         )
-        val samplePath = tempFolder.newFile("sample-file").toPath()
+        val programPath = makeSampleCliExecutable()
         runInEdtAndWait {
-            val execCommand = RunCommandDialog(projectRule.project, container).buildExecCommandConfiguration(command, samplePath)
-            assertThat(execCommand.name).isEqualTo("sample-container")
-            assertThat(execCommand.parameters).isEqualTo(verifyCommand)
-            assertThat(execCommand.isUseConsole).isTrue
-            assertThat(execCommand.isShowConsoleOnStdOut).isTrue
-            assertThat(execCommand.program).isEqualTo(samplePath.toString())
+            val environmentVariables = mutableListOf<String>()
+            val environment =
+                ExecutionEnvironmentBuilder
+                    .create(
+                        projectRule.project,
+                        DefaultRunExecutor.getRunExecutorInstance(),
+                        RunCommandRunProfile(
+                            connectionSettings?.toEnvironmentVariables(),
+                            RunCommandDialog(projectRule.project, container, connectionSettings)
+                                .constructExecCommandParameters(command),
+                            containerName, programPath.toAbsolutePath().toString()
+                        )
+                    )
+                    .build {
+                        it.processHandler?.addProcessListener(object : ProcessAdapter() {
+                            override fun onTextAvailable(event: ProcessEvent, outputType: Key<*>) {
+                                super.onTextAvailable(event, outputType)
+                                environmentVariables.add(event.text.split("\n").first())
+                            }
+
+                            override fun processTerminated(event: ProcessEvent) {
+                                super.processTerminated(event)
+                                assertThat(environmentVariables[1]).isEqualTo("AccessKeyEcsExecDummy")
+                                assertThat(environmentVariables[2]).isEqualTo("SecretKeyEcsExecDummy")
+                                assertThat(environmentVariables[3]).isEqualTo(dummyRegion)
+                                assertThat(environmentVariables[4]).isEqualTo(dummyRegion)
+                            }
+                        })
+                    }
```

Comment:
Do these actually fail the test? Or is this thrown on a different thread so the test ignores it?

Could instead use a CapturingProcessAdaptor and then make the asserts on the outputs

May want a CountDownLatch and an await after the execute to make sure the test does not exit before the process exists too

Reply:
Hadn't changed the region in the credentials, so they failed and were not in the same thread so the tests were displayed as passed. A countdown latch can be used with a specific count (5) and without the CapturingProcessAdapter